stan/math/rev/core/count_vars.hpp

```diff
@@ -0,0 +1,157 @@
+#ifndef STAN_MATH_REV_CORE_COUNT_VARS_HPP
+#define STAN_MATH_REV_CORE_COUNT_VARS_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/functor.hpp>
+#include <stan/math/rev/fun.hpp>
```

Comment:
idk if we want to hopscotch back and forth between core and fun, or is that normal?

Reply:
My understanding is that fun is the meat that calls meta / core

Reply:
Good call.